src/main/java/de/mediathekview/mserver/crawler/wdr/parser/WdrVideoJsonDeserializer.java

```diff
@@ -0,0 +1,148 @@
+  private static final String ATTRIBUTE_CAPTION_URL = "captionURL";
+  private static final String ATTRIBUTE_SIGN_LANUAGE = "slVideoURL";
+  private static final String ATTRIBUTE_VERSION = "mediaVersion";
+  private static final String ATTRIBUTE_VIDEO_URL = "videoURL";
+  private static final String ATTRIBUTE_XML = "xml";
+
+  private static final String MEDIA_VERSION_1_1 = "1.1.0";
+  private static final String MEDIA_VERSION_1_2 = "1.2.0";
+  private static final String MEDIA_VERSION_1_3 = "1.3.0";
+  
+  private final String protocol;
+  
+  public WdrVideoJsonDeserializer(final String aProtocol) {
+    protocol = aProtocol;
+  }
+  
+  @Override
+  public Optional<WdrMediaDTO> deserialize(JsonElement aJsonElement, Type aType, JsonDeserializationContext aContext) throws JsonParseException {
+    JsonObject jsonObject = aJsonElement.getAsJsonObject();
+    
+    Optional<String> version = JsonUtils.getAttributeAsString(jsonObject, ATTRIBUTE_VERSION);
+    if (version.isPresent()) {
+      switch(version.get()) {
+        case MEDIA_VERSION_1_1:
+          return deserializeVersion1_1(jsonObject);
+        case MEDIA_VERSION_1_2:
+          return deserializeVersion1_2(jsonObject);
+        case MEDIA_VERSION_1_3:
+          return deserializeVersion1_3(jsonObject);
+        default:
+          LOG.error("WdrVideoJsonDeserializer: unsupported media version: " + version.get());
+      }
+    }
+    
+    return Optional.empty();
+  }  
+
+  private Optional<WdrMediaDTO> deserializeVersion1_1(final JsonObject aJsonObject) {
+    
+    if(aJsonObject.has(ELEMENT_MEDIA_RESOURCE)) {
+      final JsonObject mediaResourceObject = aJsonObject.get(ELEMENT_MEDIA_RESOURCE).getAsJsonObject();
+      
+      if (mediaResourceObject.has(ELEMENT_ALT)) {
+        final JsonObject dfltObject = mediaResourceObject.get(ELEMENT_ALT).getAsJsonObject();
+        Optional<String> mp4Url = JsonUtils.getAttributeAsString(dfltObject, ATTRIBUTE_VIDEO_URL);
+
+        return createMediaDTO(mp4Url, Optional.empty(), Optional.empty(), Optional.empty());
+      }
+    }
+    
+    return Optional.empty();
+  }
+  
+  private Optional<WdrMediaDTO> deserializeVersion1_2(final JsonObject aJsonObject) {
+    
+    if(aJsonObject.has(ELEMENT_MEDIA_RESOURCE)) {
+      final JsonObject mediaResourceObject = aJsonObject.get(ELEMENT_MEDIA_RESOURCE).getAsJsonObject();
+      Optional<String> subtitleUrl = JsonUtils.getAttributeAsString(mediaResourceObject, ATTRIBUTE_CAPTION_URL);
+      
+      if (mediaResourceObject.has(ELEMENT_DFLT)) {
+        final JsonObject dfltObject = mediaResourceObject.get(ELEMENT_DFLT).getAsJsonObject();
+        Optional<String> m3u8Url = JsonUtils.getAttributeAsString(dfltObject, ATTRIBUTE_VIDEO_URL);
+
+        return createMediaDTO(m3u8Url, subtitleUrl, Optional.empty(), Optional.empty());
+      }
+    }
+    
+    return Optional.empty();
+  }
+  
+  private Optional<WdrMediaDTO> deserializeVersion1_3(final JsonObject aJsonObject) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)